analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TypeExaminationOnSystemType.CSharp12.cs

```diff
@@ -0,0 +1,7 @@
+﻿using System;
+using Point2D = (int, int);
+
+Point2D p = new(1, 2);
+
+if (p.GetType().IsInstanceOfType(typeof(Point2D))) //Noncompliant
+{ /* ... */ }
```

Comment:
Optional: I would also test a couple of compliant cases: 
- `typeof(Point2D).IsInstanceOfType("42")`
- `typeof(Point2D).IsInstanceOfType(typeof(Point2D))`